modules/tensor_mechanics/src/materials/cohesive_zone_model/CZMRealVectorScalar.C

```diff
@@ -0,0 +1,69 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CZMRealVectorScalar.h"
+#include "CohesiveZoneModelTools.h"
+
+registerMooseObject("TensorMechanicsApp", CZMRealVectorScalar);
+
+InputParameters
+CZMRealVectorScalar::validParams()
+{
+  InputParameters params = InterfaceMaterial::validParams();
+  params.addClassDescription(
+      "Compute a scalar value from a RealVector defined on a cohesive zone interface");
+  params.addRequiredParam<std::string>("real_vector_value", "The vector material name");
+  params.addRequiredParam<MaterialPropertyName>(
+      "property_name", "Name of the material property computed by this model");
+  MooseEnum scalarType("Normal Tangent");
+  params.addRequiredParam<MooseEnum>(
+      "scalar_type", scalarType, "Type of scalar to be computed: Normal, Tangent");
```

Comment:
we probably need a better naming. maybe `direction_type` or something else.

There will be two tangential directions and how do you want to output them separately.

Reply:
for the tangent direction, we only output the norm of the tangent vector. Knowing the tangent components in interface coordiante is uselles, because the two tangent directions vary elemnt by element.

Reply:
I will change this to `direction`